course/page/inline.py

```diff
@@ -174,23 +175,19 @@ def get_weight(self, answer):
             return 0
 
     def get_field_layout(self, correctness=None):
+        kwargs = {}
+        kwargs["prepended_text"] = getattr(self.answers_desc, "prepended_text", "")
+        kwargs["appended_text"] = getattr(self.answers_desc, "appended_text", "")
+        kwargs["use_popover"] = "true"
+        kwargs["popover_title"] = getattr(self.answers_desc, "hint_title", "")
+        kwargs["popover_content"] = getattr(self.answers_desc, "hint", "")
         if correctness is None:
-            return Field(
-                    self.name,
-                    use_popover="true",
-                    popover_title=getattr(self.answers_desc, "hint_title", ""),
-                    popover_content=getattr(self.answers_desc, "hint", ""),
-                    style=self.get_width_str()
-                    )
+            kwargs["style"] = self.get_width_str()
         else:
-            return Field(
-                    self.name,
-                    use_popover="true",
-                    popover_title=getattr(self.answers_desc, "hint_title", ""),
-                    popover_content=getattr(self.answers_desc, "hint", ""),
-                    style=self.get_width_str(self.width + 2),
-                    correctness=correctness
-                    )
+            kwargs["style"] = self.get_width_str(self.width + 2)
+            kwargs["correctness"] = correctness
+
+        return PrependedAppendedText(self.name, **kwargs)
```

Comment:
Yes, according to the definition of ``PrependedAppendedText``, it is a subclass of Field with ``prepended_text/appended_text=None`` as optional arguements.
```
class PrependedAppendedText(Field):
    template = "%s/layout/prepended_appended_text.html"

    def __init__(self, field, prepended_text=None, appended_text=None, *args, **kwargs):
        self.field = field
        self.appended_text = appended_text
        self.prepended_text = prepended_text
        if 'active' in kwargs:
            self.active = kwargs.pop('active')

        self.input_size = None
        css_class = kwargs.get('css_class', '')
        if 'input-lg' in css_class:
            self.input_size = 'input-lg'
        if 'input-sm' in css_class:
            self.input_size = 'input-sm'

        super(PrependedAppendedText, self).__init__(field, *args, **kwargs)
```

What's more, I implemented that in the ``AnswerBase``  as I had thought that it can also be used for ``ChoicesAnswer``, and I did try to add ``prepended_text`` and ``appended_text`` also to the ``allowed_attr`` for ``ChoicesAnswer`` question, but I then removed them as I found it has been suggested not doing so by Bootstrap docs[http://getbootstrap.com/components/#input-groups]:
>
## Textual `<input>`s only
Avoid using `<select>` elements here as they cannot be fully styled in WebKit browsers.
Avoid using `<textarea>` elements here as their rows attribute will not be respected in some cases.
>

Shall I revert the revision of  ``get_field_layout`` back in ``AnswerBase``  and make a implementation for ``ShortAnswer``? I felt it unnecessary according to the declaration of ``PrependedAppendedText``.